flaml/autogen/agent/math_user_proxy_agent.py

```diff
@@ -0,0 +1,462 @@
+        name="MathChatAgent",  # default set to MathChatAgent
+        system_message="",
+        work_dir=None,
+        human_input_mode="NEVER",  # Fully automated
+        max_consecutive_auto_reply=None,
+        is_termination_msg=is_termination_msg,
+        use_docker=True,
+        max_invalid_q_per_step=3,  # a parameter needed in MathChat
+        **config,
+    ):
+        """
+        Args:
+            name (str): name of the agent
+            system_message (str): system message to be sent to the agent
+            work_dir (str): working directory for the agent
+            human_input_mode (str): whether to ask for human inputs every time a message is received.
+                Possible values are "ALWAYS", "TERMINATE", "NEVER".
+                (1) When "ALWAYS", the agent prompts for human input every time a message is received.
+                    Under this mode, the conversation stops when the human input is "exit",
+                    or when is_termination_msg is True and there is no human input.
+                (2) When "TERMINATE", the agent only prompts for human input only when a termination message is received or
+                    the number of auto reply reaches the max_consecutive_auto_reply.
+                (3) When "NEVER", the agent will never prompt for human input. Under this mode, the conversation stops
+                    when the number of auto reply reaches the max_consecutive_auto_reply or when is_termination_msg is True.
+            max_consecutive_auto_reply (int): the maximum number of consecutive auto replies.
+                default to None (no limit provided, class attribute MAX_CONSECUTIVE_AUTO_REPLY will be used as the limit in this case).
+                The limit only plays a role when human_input_mode is not "ALWAYS".
+            is_termination_msg (function): a function that takes a message and returns a boolean value.
+                This function is used to determine if a received message is a termination message.
+            use_docker (bool): whether to use docker to execute the code.
+            max_invalid_q_per_step (int): (ADDED) the maximum number of invalid queries per step.
+            **config (dict): other configurations.
+        """
+        super().__init__(
+            name=name,
+            system_message=system_message,
+            work_dir=work_dir,
+            human_input_mode=human_input_mode,
+            max_consecutive_auto_reply=max_consecutive_auto_reply,
+            is_termination_msg=is_termination_msg,
+            use_docker=use_docker,
+            **config,
+        )
+
+        # fixed var
+        self._max_invalid_q_per_step = max_invalid_q_per_step
+
+        # mutable
+        self._valid_q_count = 0
+        self._total_q_count = 0
+        self._accum_invalid_q_per_step = 0
+        self._previous_code = ""
+        self.last_reply = None
+
+    def generate_prompt(self, problem, prompt_type="default", customized_prompt=None):
```

Comment:
```suggestion
    def generate_init_prompt(self, problem, prompt_type="default", customized_prompt=None):
```
and check the places where this function is used.